### ISSUE TITLE
Tested/Used firmeware and devices

### ISSUE THREAD

**Issue by frawe** [frawe/EasyControls3_homeassistant#9]
This issue should be used to collect supported firmware versions and devices.

I have a KWL 360 W ET running currently 1.0.19.

**Comment by zsolth** [comment]
KWL 500 W L on 1.0.9

**Comment by zsolth** [comment]
> This issue should be used to collect supported firmware versions and devices.
> 
> I have a KWL 360 W ET running currently 1.0.19.

Super sorry to hijack this thread here, but I figrued I'd ask, is the enthalpy heat exchanger worth investing in? We're using the 500W (with basic HE) and relative humidity gets up to 70%  in the summer (as we're colling with surface cooling that doesn't solve humidity issues like an AC) and down to 30% in the winter and was wondering if the enalphy component (which can be retrofitted, I've checked) would solve this problem (or at least make the situation less bad)...

**Comment by frawe** [comment]
maybe we could switch over to https://community.home-assistant.io/t/re-helios-easycontrol-3-0-to-homeassistant-mqtt/778285/5 (you should be able to find me there and write me a message)

or also post in the public thread

then others also can find it and read it - and this can be keeped clean ;)

**Comment by sanchosk** [comment]
KWL 300 W ET L
1.0.17

Minor issue here - the HA only recognized 4 entities...
![image](https://github.com/user-attachments/assets/32312523-0fd6-4dd0-a802-975c037707b1)

**Comment by frawe** [comment]
That is sad to here 😢 

Did you add the integration newly or updated?

Did you also tried to restart home assistant?

**Comment by sanchosk** [comment]
Installed using adding repo to HACS, setting version to 1.3, restarted HA after install, provided IP address (did not worked with DNS, reported unable to connect). Did not tried to restart after that, but I can give it a go.

**Comment by sanchosk** [comment]
Hey!
I was skeptical, but a promise is a promise. And voila!
![image](https://github.com/user-attachments/assets/42b49d1a-c400-4958-8ee3-a79b53df0f72)

**Comment by frawe** [comment]
great to see 😄 

i will look into that, to see if I can find a reason for that...

**Comment by kultikom** [comment]
KWL 300 WL 
Firmware 1.0.17

works fine, but had to restart HA after installation to see sensors, as mentioned above.

**Comment by mgrfilipmarek** [comment]
KWL 360 W
Firmware 1.0.17
Firmware 1.0.23
Everything works fine.

**Comment by EagleIJoe** [comment]
KWL 170 W
Firmware 1.0.19
Everything works fine

**Comment by soska12** [comment]
KWL 300 W ET L
1.0.23
Looking good so far

**Comment by zsolth** [comment]
Few months ago I also updated to 1.0.23 for KWL 500 ET L, so far so good

**Comment by indivliving** [comment]
KWL 470 W (1.0.23) -> not working
KWL 250 W (1.0.23) -> working

**Comment by frawe** [comment]
> KWL 470 W (1.0.23) -> not working KWL 250 W (1.0.23) -> working

sad to here, can you mabe give more details, what is the probalem with the KWL 470 W?

**Comment by indivliving** [comment]
just a red banner with "unknown" when im trying to add

**Comment by frawe** [comment]
Ok, but your home assistant server is in the same net as your KWL? And you can reach the webserver and so on? If that, then maybe write me in the home assistant forum and I try to see if I can help some how 😄

**Comment by indivliving** [comment]
Yes, same Net and easyControl Webserver is working. Ok, will write you. Thank you

**Comment by Sagittarius4Star** [comment]
Cheers!

Just to report that KWL 500 W ET L on v1.0.23 does not work.
Cannot be added via:
. IP only = error "unknown"
. URL with http://ip = "cannot_connect"

**Comment by Sagittarius4Star** [comment]
FYI; if "device_type" string gets ignored (e.g. commented out), URL gets accepted and is working fine.

**Comment by TheRoby** [comment]
Hello,
I have a KWL 470 W ET, with 1.0.23 and it's not working ...
@Sagittarius4Star what do you mean by " if "device_type" string gets ignored (e.g. commented out)" ? where ?
Thank you :)

**Comment by frawe** [comment]
hi, 

sad to here that.

Could someone of you (meaning persons where it doesn't work) pleas try the following and report it back to me.

After the integration was added via hacs download, go to the folder where the files are stored (custom_components/EasyControls3_homeassistant/EasyControls3Instance.py) and modify the file to look like that:

`    def _parseData(self, data):
        # device info
        try:
            self._deviceModel = deviceInfo["device_model_data"][data[17 * 2 + 1]]
        except IndexError:
            self._deviceModel = deviceInfo["device_model_data"][0]

        try:
            self._deviceType = deviceInfo["device_type_data"][data[16 * 2 + 1]]
        except IndexError:
            self._deviceType = deviceInfo["device_type_data"][0]

        self._SerialNR = (
`

**Comment by ewigbauer** [comment]
Hi, I had the same problem with the KWL250!

Then I also commented out line 76 and lo and behold, it works! Thanks for the tip.

Silvio

**Comment by TheRoby** [comment]
Hi,
It's ok for me and my KWL470ET (with the commented out line 76, or the "try" ;)

Thanks

**Comment by TheRoby** [comment]
Oups, no, sorry, the "try" version is not working after a full restart of HA ...

**Comment by DerDanilo** [comment]
Hi @frawe,

first of all — thank you for the work you've put into this integration.
It's been the foundation everything else builds on, and on my Helios
KWL 250 W (firmware 1.0.25) it has been running reliably for quite a
while. Much appreciated.

In the meantime I've extended your integration into my own fork and
published it as a separate community project:

https://github.com/DerDanilo/homeassistant_helios_easycontrol3

A few things that have been added on top of your v1.3.0:

- **Bounds-checked device-model / device-type lookups** — bytes that
  aren't in the bundled list fall back to `"unknown (N)"` instead of
  raising an `IndexError` during the config flow (relevant for the
  issue reported here).
- **DataUpdateCoordinator pattern** — one polling cycle for all entities
  instead of per-entity polling.
- **Snapshot persistence** — entities keep their last good value during
  brief connection outages instead of going `unavailable`.
- **Always-available reachability sensor** (`binary_sensor.*_kwl_reachable`)
  + connection-stability % over a rolling 60-minute window + failure
  counters, so you can alert on KWL downtime.
- **More entities** — real fan RPMs, supply-air target setpoints per
  mode, individual humidity / CO2 sensor readings, IO state diagnostics,
  last power interruption timestamp, combined operating time.
- **Cool Recovery with a plug-confirmation safety flow** that mirrors
  the WebUI safety prompt.
- **Vacation mode and Weekly Schedule on/off switches** (programming
  itself stays in the Helios WebUI).
- **English + German UI translations** for all entity names.
- HACS-compatible as a custom repository.

It's a separate integration domain (`helios_easycontrol3`), so it can be
installed side-by-side with this one without conflicting.

Tested only against my own KWL 250 W on firmware 1.0.25, so anyone with
a different model / firmware combination is very welcome to open an
issue on the fork with their results.

Not trying to redirect attention away from this project — just sharing
in case the fork helps unblock anyone here while a fix lands upstream.
Happy to coordinate or contribute back if useful.

**Comment by frawe** [comment]
@DerDanilo 
I really appreciate your work — and it’s great that you even created a proper HACS integration. That was on my to‑do list as well, but with a small child (getting constantly sick - I am also getting sick) and a house that’s still not finished, I just haven’t had the time to get to it.

So thank you for putting in the effort.

One thing I was wondering, though: why didn’t you open a pull request? We could have improved the repository together. Was there a particular reason you decided to create a new repo instead of considering a collaboration? If you want to talk you also can conntact me via https://community.home-assistant.io I have the same user name   (fr4w3) there.

**Comment by DerDanilo** [comment]
I'll write you via the HA Community once I find some time. I had pretty similar motivations around family and our house setup, but it still took me over a year before I finally sat down and tackled it properly.

Regarding the new repo: the main reason was that I wanted a cleaner and more logical repository name that also fits better into the HACS naming/domain conventions. The changes also became a bit more extensive over time.

Another factor was simply the turnaround time. Sometimes it takes quite a while until somebody responds, and with larger merge requests it can easily turn into weeks or even longer. I completely understand that maintainers have limited time, but for me it became difficult to iterate on larger structural changes that way.

That said, I'm still happy to create a merge request for your repo if you'd like. I assume you'd probably want to adjust parts of the README and maybe some surrounding details first though. We can sort those things out via HA Community PMs then.

Alternatively, you could also just fork/clone my repo and we continue development there together. The naming is already aligned properly for HACS conventions, and I think that would also give us a cleaner path towards eventually getting it into the official HACS repositories.
